main.go

```diff
@@ -88,6 +89,8 @@ func main() {
 	flag.DurationVar(&requeueDependency, "requeue-dependency", 30*time.Second, "The interval at which failing dependencies are reevaluated.")
 	flag.BoolVar(&watchAllNamespaces, "watch-all-namespaces", true,
 		"Watch for custom resources in all namespaces, if set to false it will only watch the runtime namespace.")
+	flag.BoolVar(&noRemoteBases, "no-remote-bases", false,
```

Comment:
It may be beneficial to convey the security implications on the flag name (e.g. `--insecure-remote-bases`).

Now compounding the security and the performance impact, it would be good to consider this to be disabled by default, specially ahead of GA.
Alternatively, we could add a flag to `flux cli`, in which we always enforce the best security best practices across all controllers for a given Flux version. But that flag would not promise/honour backwards compatibility.

Reply:
I'm for including the `--no-remote-bases=true` into the security profile as described in [[RFC] Flux Multi-Tenancy Mode](https://github.com/fluxcd/flux2/pull/2093).

Reply:
Let's add this feature as is, and discuss on a separate forum about how to enable security focused users to adopt breaking changes more seamlessly - as well as the MT topics.